jerry-core/vm/vm.c

```diff
@@ -1984,6 +1966,24 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
           *stack_top_p++ = result;
           continue;
         }
+        case VM_OC_RESOLVE_BASE_FOR_CALL:
+        {
+          ecma_value_t this_value = stack_top_p[-3];
+
+          if (this_value == ECMA_VALUE_REGISTER_REF)
+          {
+            /* Lexical environment cannot be 'this' value. */
+            stack_top_p[-2] = ECMA_VALUE_UNDEFINED;
+            stack_top_p[-3] = ECMA_VALUE_UNDEFINED;
```

Comment:
Is this assignment really needed here?

Reply:
Tl;dr : **Yes**.

Explanation:

If this new bytecode has been emitted, the previous bytecode is always `CBC_PUSH_IDENT_REFERENCE`. 

If the condition evaluates to true, the next operations happened in the previous vm_instruction:

```
*stack_top_p++ = ECMA_VALUE_REGISTER_REF; 
*stack_top_p++ = literal_index;
*stack_top_p++ = ecma_fast_copy_value (frame_ctx_p->registers_p[literal_index]);
``` 
So that's why `stack_top_p[-3]` is `ECMA_VALUE_REGISTER_REF`. When the function call is performed and the arguments have been "popped" from the stack, the `stack_top_p[-3]` represents the `this_value` during the call. Since no more specification is needed `ECMA_VALUE_UNDEFINED` is a good solution for it.

Also, `stack_top[-2]` is a current literal index, which is a simple number not a valid ecma-value. Change it to `ECMA_VALUE_UNDEFINED` makes it a _valid and not costly_ value.